crowbar_framework/app/models/service_object.rb

```diff
@@ -1540,6 +1524,79 @@ def wait_for_chef_daemons(node_list)
 
   private
 
+  THREAD_POOL_SIZE = 20
+
+  def release_chef_locks(locks)
+    return if locks.empty?
+
+    queue = Queue.new
+    locks.each { |l| queue.push l }
+
+    workers = (1...THREAD_POOL_SIZE).map do
```

Comment:
Oh I am ruby-blind. I see now the inner loop.

Still I do not get why to do that. The locks are release into a thread, but the main thread is still blocked in the `.join()`, so there is not performance improvement.

Why not to collect that in the main thread then?